File: python/baiduads-sdk/test/oauth/oauth_service_test.py
"""
for test OAuthService
"""
import unittest

from baiduads_manual.oauth.api.oauth_service import OAuthService
from baiduads_manual.oauth.model.get_access_token_request import GetAccessTokenRequest
from baiduads_manual.oauth.model.get_open_id_request import GetOpenIdRequest
from baiduads_manual.oauth.model.get_user_info_request import GetUserInfoRequest
from baiduads_manual.oauth.model.refresh_token_request import RefreshTokenRequest


class TestOAuthService(unittest.TestCase):

    def setUp(self):
        self.api = OAuthService()  # noqa: E501
        self.authCode = "xxxxx"
        self.secretKey = "xxxxx"
        self.appId = "xxxxx"
        self.userId = 0
        self.grantType = "access_token"

    def tearDown(self):
        pass

    def test_get_access_token1(self):
        """Test case for get_access_token

        测试异常情况
        """
        request = GetAccessTokenRequest("x", "x", "x", "x", 1)
        res = self.api.get_access_token(request)
        assert res.code == 30002
        assert res.message == "应用不存在"

    def test_get_access_token2(self):
        """Test case for  get_access_token
        """
        request = GetAccessTokenRequest(self.appId,
                                        self.authCode,
                                        self.secretKey,
                                        self.grantType,
                                        self.userId)
        res = self.api.get_access_token(request)
        assert res.code == 0
        assert res.data.accessToken is not None
        assert res.data.refreshToken is not None

    def test_get_open_id1(self):
        request = GetOpenIdRequest("x", 1)
        res = self.api.get_open_id(request)
        assert res.code == 50004

    def test_get_open_id2(self):
        request = GetAccessTokenRequest(self.appId,
                                        self.authCode,
                                        self.secretKey,
                                        self.grantType,
                                        self.userId)
        res = self.api.get_access_token(request)
        accessToken = res.data.accessToken

        request = GetOpenIdRequest(accessToken, self.userId)
        res = self.api.get_open_id(request)
        assert res.code == 0
        assert res.data.openId is not None

    def test_refresh_token1(self):
        request = RefreshTokenRequest("x", "x", "x", 1)
        res = self.api.refresh_token(request)
        assert res.code == 50004
        assert res.message == "token验证不通过"

    def test_refresh_token2(self):
        request = GetAccessTokenRequest(self.appId,
                                        self.authCode,
                                        self.secretKey,
                                        self.grantType,
                                        self.userId)
        res = self.api.get_access_token(request)
        refreshToken = res.data.refreshToken

        request = RefreshTokenRequest(self.appId,
                                      refreshToken,
                                      self.secretKey,
                                      self.userId)
        res = self.api.refresh_token(request)
        assert res.code == 0
        assert res.data.accessToken is not None
        self.access_token = res.data.accessToken

    def test_get_user_info1(self):
        request = GetAccessTokenRequest(self.appId,
                                        self.authCode,
                                        self.secretKey,
                                        self.grantType,
                                        self.userId)
        res = self.api.get_access_token(request)
        accessToken = res.data.accessToken

        request = GetOpenIdRequest(accessToken, self.userId)
        res = self.api.get_open_id(request)
        openId = res.data.openId

        request = GetUserInfoRequest(accessToken,
                                     self.userId,
                                     openId)
        res = self.api.get_user_info(request)
        assert res.code == 0
        assert res.data.masterUid is not None

    def test_get_user_info2(self):
        request = GetUserInfoRequest("x",
                                     self.userId,
                                     "x")
        res = self.api.get_user_info(request)
        assert res.code == 50004


if __name__ == '__main__':
    unittest.main()

File: java/baiduads-sdk/src/test/com/baidu/dev2/api/sdk/oauth/OAuthServiceTest.java
package com.baidu.dev2.api.sdk.oauth;


import com.baidu.dev2.api.sdk.invoke.ApiException;
import com.baidu.dev2.api.sdk.manual.oauth.api.OAuthService;
import com.baidu.dev2.api.sdk.manual.oauth.model.GetAccessTokenRequest;
import com.baidu.dev2.api.sdk.manual.oauth.model.GetAccessTokenResponse;
import com.baidu.dev2.api.sdk.manual.oauth.model.GetOpenIdRequest;
import com.baidu.dev2.api.sdk.manual.oauth.model.GetOpenIdResponse;
import com.baidu.dev2.api.sdk.manual.oauth.model.GetUserInfoRequset;
import com.baidu.dev2.api.sdk.manual.oauth.model.GetUserInfoResponse;
import com.baidu.dev2.api.sdk.manual.oauth.model.RefreshTokenRequest;
import com.baidu.dev2.api.sdk.manual.oauth.model.RefreshTokenResponse;
import org.junit.Assert;
import org.junit.Before;
import org.junit.Test;

public class OAuthServiceTest {
    private OAuthService oAuthService;

    private String appId = "xxxxx";
    private Long userId = 1L;
    private String authCode = "xxxxx";
    private String secretKey = "xxxxx";
    private String accessToken;
    private String refreshToken;
    private String openId;

    @Before
    public void before() {
        oAuthService = new OAuthService();
    }

    @Test
    public void getAccessToken1() throws ApiException {
        GetAccessTokenRequest request = new GetAccessTokenRequest();
        request.setAppId(appId);
        request.setUserId(userId);
        request.setAuthCode(authCode);
        request.setSecretKey(secretKey);
        GetAccessTokenResponse response = oAuthService.getAccessToken(request);
        Assert.assertTrue(response.getCode() == 0);
        Assert.assertNotNull(response.getData());
        Assert.assertNotNull(response.getData().getAccessToken());
        Assert.assertNotNull(response.getData().getRefreshToken());
    }

    @Test
    public void getAccessToken2() throws ApiException {
        GetAccessTokenRequest request = new GetAccessTokenRequest();
        request.setAppId("x");
        request.setUserId(userId);
        request.setAuthCode(authCode);
        request.setSecretKey(secretKey);
        GetAccessTokenResponse response = oAuthService.getAccessToken(request);
        Assert.assertTrue(response.getCode() == 30002);

    }

    @Test
    public void getOpenId() throws ApiException {
        GetAccessTokenRequest req = new GetAccessTokenRequest();
        req.setAppId(appId);
        req.setUserId(userId);
        req.setAuthCode(authCode);
        req.setSecretKey(secretKey);
        GetAccessTokenResponse resp = oAuthService.getAccessToken(req);
        Assert.assertTrue(resp.getCode() == 0);
        accessToken = resp.getData().getAccessToken();
        refreshToken = resp.getData().getRefreshToken();


        GetOpenIdRequest request = new GetOpenIdRequest();
        request.setAccessToken(accessToken);
        request.setUserId(userId);
        GetOpenIdResponse response = oAuthService.getOpenId(request);
        Assert.assertTrue(response.getCode() == 0);
        Assert.assertNotNull(response.getData());
        Assert.assertNotNull(response.getData().getOpenId());
        openId = response.getData().getOpenId();
    }

    @Test
    public void getUserInfo() throws ApiException {
        GetAccessTokenRequest req = new GetAccessTokenRequest();
        req.setAppId(appId);
        req.setUserId(userId);
        req.setAuthCode(authCode);
        req.setSecretKey(secretKey);
        GetAccessTokenResponse resp = oAuthService.getAccessToken(req);
        Assert.assertTrue(resp.getCode() == 0);
        accessToken = resp.getData().getAccessToken();
        refreshToken = resp.getData().getRefreshToken();


        GetOpenIdRequest req2 = new GetOpenIdRequest();
        req2.setAccessToken(accessToken);
        req2.setUserId(userId);
        GetOpenIdResponse resp2 = oAuthService.getOpenId(req2);
        Assert.assertTrue(resp2.getCode() == 0);
        openId = resp2.getData().getOpenId();


        GetUserInfoRequset request = new GetUserInfoRequset();
        request.setUserId(userId);
        request.setAccessToken(accessToken);
        request.setOpenId(openId);
        GetUserInfoResponse response = oAuthService.getUserInfo(request);
        Assert.assertTrue(response.getCode() == 0);
        Assert.assertNotNull(response.getData());
        Assert.assertNotNull(response.getData().getMasterUid());
    }


    @Test
    public void refreshToken() throws ApiException {
        GetAccessTokenRequest req = new GetAccessTokenRequest();
        req.setAppId(appId);
        req.setUserId(userId);
        req.setAuthCode(authCode);
        req.setSecretKey(secretKey);
        GetAccessTokenResponse resp = oAuthService.getAccessToken(req);
        Assert.assertTrue(resp.getCode() == 0);
        accessToken = resp.getData().getAccessToken();
        refreshToken = resp.getData().getRefreshToken();

        RefreshTokenRequest request = new RefreshTokenRequest();
        request.setRefreshToken(refreshToken);
        request.setAppId(appId);
        request.setUserId(userId);
        request.setSecretKey(secretKey);
        RefreshTokenResponse response = oAuthService.refreshToken(request);
        Assert.assertTrue(response.getCode() == 0);
        Assert.assertNotNull(response.getData());
        Assert.assertNotNull(response.getData().getAccessToken());
        Assert.assertNotNull(response.getData().getRefreshToken());
    }
}
